### PR TITLE
SOD1/2: Add support for API Package & API Package Assignment

### PR DESCRIPTION
**Pull request by IgorNekhoda**
https://github.com/abapGit/abapGit/issues/6315
https://github.com/abapGit/abapGit/issues/6316

### CODE REVIEW

File: src/objects/zcl_abapgit_object_sod1.clas.abap [hunk 1]
```diff
@@ -0,0 +1,284 @@
...
+    rv_user = c_user_unknown.
+
+  ENDMETHOD.
+
+
+  METHOD zif_abapgit_object~delete.
+
+    TRY.
+
+        DATA(lo_factory) = cl_wb_object_operator_factory=>get_object_operator( object_type = VALUE #( objtype_tr = ms_item-obj_type )
```
Comment (larshp) [inline src/objects/zcl_abapgit_object_sod1.clas.abap:46]: the code must be 702 syntax compatible

**Comment by mbtools** [comment]
Hi Igor,

in addition to 702 syntax (i.e. no in-line declaration like `DATA(...)` , no `VALUE` definitions), you have to change classes/types that do not exist in 702 to dynamic calls (like `cl_wb_object_operator_factory`). 

I suggest using [`zcl_abapgit_object_srvd`](https://github.com/abapGit/abapGit/blob/main/src/objects/zcl_abapgit_object_srvd.clas.abap) as an example.

**Comment by IgorNekhoda** [comment]
> I suggest using [`zcl_abapgit_object_srvd`](https://github.com/abapGit/abapGit/blob/main/src/objects/zcl_abapgit_object_srvd.clas.abap) as an example.

Hello Marc,
 
this is not possible due to static implementation of get_wb_object_operator and other parts in zcl_abapgit_object_srvd. I have to duplicate the coding.
 
Best regards,
Igor

**Comment by mbtools** [comment]
That's what I meant: use the srvd code as an example for your classes

**Comment by mbtools** [comment]
Looks good. A few more 702 syntax and breaking lines that are too long. That's it.

Please create a small test case in repo https://github.com/abapGit-tests/SOD1_SOD2 as well.

**Comment by mbtools** [comment]
add the following to both classes:

```abap
  METHOD zif_abapgit_object~get_deserialize_order.
    RETURN.
  ENDMETHOD.

  METHOD zif_abapgit_object~map_filename_to_object.
    RETURN.
  ENDMETHOD.

  METHOD zif_abapgit_object~map_object_to_filename.
    RETURN.
  ENDMETHOD.
```

File: src/objects/zcl_abapgit_object_sod1.clas.xml [hunk 2]
```diff
@@ -0,0 +1,16 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
+<abapGit version="v1.0.0" serializer="LCL_OBJECT_CLAS" serializer_version="v1.0.0">
+ <asx:abap xmlns:asx="http://www.sap.com/abapxml" version="1.0">
+  <asx:values>
+   <VSEOCLASS>
+    <CLSNAME>ZCL_ABAPGIT_OBJECT_SOD1</CLSNAME>
+    <LANGU>E</LANGU>
+    <DESCRIPT>abapGit - SOD1 - Authorization object</DESCRIPT>
```
Comment (larshp) [inline src/objects/zcl_abapgit_object_sod1.clas.xml:8]: please update the description

File: src/objects/zcl_abapgit_object_sod2.clas.xml [hunk 3]
```diff
@@ -0,0 +1,16 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
+<abapGit version="v1.0.0" serializer="LCL_OBJECT_CLAS" serializer_version="v1.0.0">
+ <asx:abap xmlns:asx="http://www.sap.com/abapxml" version="1.0">
+  <asx:values>
+   <VSEOCLASS>
+    <CLSNAME>ZCL_ABAPGIT_OBJECT_SOD2</CLSNAME>
+    <LANGU>E</LANGU>
+    <DESCRIPT>abapGit - SOD2 - Authorization object</DESCRIPT>
```
Comment (larshp) [inline src/objects/zcl_abapgit_object_sod2.clas.xml:8]: please update the description

**Comment by IgorNekhoda** [comment]
> Please create a small test case in repo https://github.com/abapGit-tests/SOD1_SOD2 as well.

https://github.com/IgorNekhoda/SOD1_SOD2

**Comment by larshp** [comment]
looks like there are some fields to be cleared,

```
     <CHANGED_AT>2023-07-18T22:00:00Z</CHANGED_AT>
     <CREATED_AT>2023-07-18T22:00:00Z</CREATED_AT>
```

https://docs.abapgit.org/development-guide/serializers/serializers.html

> As code is stored in git, no usernames, timestamps, states (e.g. active/inactive), or other system-specific information should be part of the serialized object files.

**Review by larshp — APPROVED** [review]
cool 👍

**Comment by larshp** [comment]
welcome to abapGit as a contributor 🎉

**Pull request merged**